src/main/java/org/truffleruby/extra/ffi/PointerNodes.java

```diff
@@ -156,23 +161,29 @@ public DynamicObject readString(DynamicObject pointer, int length) {
     public static abstract class PointerReadStringToNullPrimitiveNode extends PrimitiveArrayArgumentsNode {
 
         @Specialization(guards = "isNullPointer(pointer)")
-        public DynamicObject readNullPointer(DynamicObject pointer) {
-            return createString(RopeConstants.EMPTY_ASCII_8BIT_ROPE);
+        public DynamicObject readNullPointer(DynamicObject pointer,
+                                             @Cached("create()") AllocateObjectNode allocateObjectNode) {
+            return allocateObjectNode.allocate(coreLibrary().getStringClass(), Layouts.STRING.build(false, false, RopeConstants.EMPTY_ASCII_8BIT_ROPE));
```

Comment:
I was thinking the same thing, it would be nice to have methods that default to commonly used parameters that explain the common string creation situations. In another pass sounds good.